src/platform/ESP32/ConfigurationManagerImpl.cpp

```diff
@@ -102,7 +102,28 @@ CHIP_ERROR ConfigurationManagerImpl::_Init()
 
 CHIP_ERROR ConfigurationManagerImpl::_GetPrimaryWiFiMACAddress(uint8_t * buf)
 {
-    return esp_wifi_get_mac(WIFI_IF_STA, buf);
+    wifi_mode_t mode;
+    esp_wifi_get_mode(&mode);
+    if ((mode == WIFI_MODE_AP) || (mode == WIFI_MODE_APSTA))
+        return MapConfigError(esp_wifi_get_mac(WIFI_IF_AP, buf));
```

Comment:
@wqx6 Can there be a situation/usecase when we will have to return STA MAC address even when the mode is AP+STA? Does it depend on what the caller uses it for? Currently, I see that it is used only by mDNS for hostname.

Reply:
I think there is no such a situation when we need STA MAC address if AP mode is active for now.